src/main/kotlin/org/rust/ide/annotator/RsSyntaxErrorsAnnotator.kt

```diff
@@ -236,15 +237,32 @@ private fun checkDot3Parameter(holder: AnnotationHolder, dot3: PsiElement?) {
 
 private fun checkValueParameter(holder: AnnotationHolder, param: RsValueParameter) {
     val fn = param.parent.parent as? RsFunction ?: return
+    val pat = param.pat
+    val isComplexPattern = when (pat) {
+        null, is RsPatWild -> false
+        is RsPatIdent -> {
+            val binding = pat.patBinding
+            binding.mutability == Mutability.MUTABLE || binding.kind is RsBindingModeKind.BindByReference || pat.at != null
+        }
```

Comment:
I forgot about macros! A macro can be used in a pattern context:

```rust
macro_rules! foo {
    () => { (a, b) };
}

trait A {
    fn foo(foo!(): (i32, i32)); // OK
}
```

Let's make a (tailrec?)recursive function `isComplexPattern` which considers macro invocations:

```suggestion
        }
        is RsPatMacro -> {
            val expansion = pat.macroCall.expansion
            expansion != null && expansion is MacroExpansion.Pat && isComplexPattern(expansion.pat)
        }
```